### ISSUE TITLE
renderBase64 returned undefined

### ISSUE THREAD

**Issue by iamdriz** [amir20/phantomjs-node#337]
I have the following code:

~~~
var phantom = require('phantom');

phantom.create(function (ph) {
	ph.createPage(function (page) {
		page.set('viewportSize', { width: 375, height: 667 });
		page.set('clipRect', { top: 0, left: 0, width: 375, height: 667 });  
		page.open("http://github.com/", function (status) {
			console.log("opened github? ", status);
			page.evaluate(function () { return document.title; }, function (result) {	  
				page.render('github.jpg', {format: 'jpeg', quality: '100'});
				console.log('Page title is ' + result);
				console.log('base64', page.renderBase64('PNG'));
				ph.exit();
			});
		});
	});
});
~~~

Which should create both a JPG of the website and return it as base64.

The JPG is created fine. However in the console, the base64 is `undefined`. Is the syntax different for the renderBase64 when using this module with phantomjs?

**Comment by CarlRosell** [comment]
Did notice the same problem. By reading the code at [shim.coffee#L110](https://github.com/sgentle/phantomjs-node/blob/master/shim.coffee#L110) I saw that it was wrapped, just like render.

This is the code I used:
```js
page.renderBase64('PNG', (content) => {
  console.log('base64', content);
});
```

Hope that helped :smile:

**Comment by iamdriz** [comment]
@CarlRosell Just tried that with:
~~~
page.renderBase64('PNG', (content) => {
	console.log('hello');
	console.log('base64', content);
});
~~~
And neither of those console logs are hit.

**Comment by alextiley** [comment]
@iamdriz Did you ever get to the bottom of this one? Having the same issue - it seems the callback is never fired.

**Comment by iamdriz** [comment]
@alextiley No I didn't. I couldn't get it to return it as base64 at all.

**Comment by r6203** [comment]
I'm having the same error.
After delaying the renderBase64 function in the shim.js file it seems that I get a result everytime.
I have no idea what may cause this issue.

**Comment by amir20** [comment]
Should be fixed now with `2.0.0-alpha` see [this](https://github.com/amir20/phantomjs-node/commit/17cc97853a6d367260426f9c1672b674756a1cd7#diff-059c70f651c27e92ead35f8aced38fe3R145) on how to use it.
